include/ews/ews.hpp

```diff
@@ -18374,6 +18557,50 @@ class basic_service final
                    "Expected at least one item");
         return response_message.items().front();
     }
+
+    resolution_set
+    resolve_names_impl(const std::string& name,
+                       const std::vector<folder_id>& parent_folder_ids,
+                       search_scope scope)
+    {
+        auto version = get_request_server_version();
+        std::stringstream sstr;
+        sstr << "<m:ResolveNames "
+             << "ReturnFullContactData=\""
+             << "true"
+             << "\" "
+             << "SearchScope=\"" << internal::enum_to_str(scope) << "\" ";
+
+        if (version == server_version::exchange_2010_sp2 ||
+            version == server_version::exchange_2013 ||
+            version == server_version::exchange_2013_sp1)
+        {
+            sstr << "ContactDataShape=\"IdOnly\"";
+        }
+        sstr << ">";
+        if (parent_folder_ids.size() > 0)
+        {
+            sstr << "<ParentFolderIds>";
+            for (const auto& id : parent_folder_ids)
+            {
+                sstr << id.to_xml();
+            }
+            sstr << "</ParentFolderIds>";
+        }
+        sstr << "<m:UnresolvedEntry>" << name << "</m:UnresolvedEntry>"
+             << "</m:ResolveNames>";
+        auto response = request(sstr.str());
+        const auto response_message =
+            internal::resolve_names_response_message::parse(std::move(response));
+        if (response_message.result().code ==
+                response_code::error_name_resolution_no_results ||
+            response_message.result().code ==
+                response_code::error_name_resolution_no_mailbox)
+        {
+            return resolution_set();
+        }
```

Comment:
Nonetheless, you need to throw when you encounter an error other than `error_name_resolution_no_results` and `error_name_resolution_no_mailbox` here.